packages/amplify-graphql-api-construct-tests/src/__tests__/conversations/conversation.test.ts

```diff
@@ -90,21 +96,41 @@ describe('conversation', () => {
         });
 
         // send a message to the conversation
-        const sendMessageResult = await doSendMessagePirateChat(apiEndpoint, accessToken, conversationId, [{ text: 'Hello, world!' }]);
+        const sendMessageResult = await doSendMessagePirateChat({
+          apiEndpoint,
+          accessToken,
+          conversationId,
+          content: [{ text: 'Hello, world!' }],
+        });
         const message = sendMessageResult.body.data?.pirateChat;
         expect(message).toBeDefined();
         expect(message.content).toHaveLength(1);
         expect(message.content[0].text).toEqual('Hello, world!');
         expect(message.conversationId).toEqual(conversationId);
 
+        const events: ConversationMessageStreamPart[] = [];
         // expect to receive the assistant response in the subscription
         for await (const event of subscription) {
-          expect(event.onCreateAssistantResponsePirateChat.conversationId).toEqual(conversationId);
-          expect(event.onCreateAssistantResponsePirateChat.content).toHaveLength(1);
-          expect(event.onCreateAssistantResponsePirateChat.content[0].text).toBeDefined();
-          expect(event.onCreateAssistantResponsePirateChat.role).toEqual('assistant');
-          break;
+          events.push(event.onCreateAssistantResponsePirateChat);
+          if (event.onCreateAssistantResponsePirateChat.stopReason) break;
         }
+
+        // reconstruct the message from the events
+        const sortedEvents = events
+          .filter((event) => event.contentBlockText)
+          .sort((a, b) => a.contentBlockDeltaIndex - b.contentBlockDeltaIndex);
```

Comment:
It looks like we are able to assume that all of the events from from this single subscription have the same `contentBlockIndex`? I'm curious what would happen if I sent two messages at the same time with the same `conversationID` e.g. does one fail or does one hang or is it undefined?

Reply:
> It looks like we are able to assume that all of the events from from this single subscription have the same contentBlockIndex? 

Close, but not quite.
The `contentBlockIndex` represents the index of a content block for a given assistant response message. In the example below, the first content block has a `contentBlockIndex` of 0, the second 1.
```js
{
  // other fields
  content: [
    {
       text: "Checking the weather",
    },
    {
       text: "The temperature in Charleston, SC is 84° F currently",
    },
  ]
}
```

This test is indeed assuming that there will only be one content block returned from the assistant because the test implementation guarantees it. 

In hindsight, this test should just be using the [reconcileStreamEvents](https://github.com/aws-amplify/amplify-category-api/pull/2986/files/51ae7fc76998e5901b12c33fe8666fbfb607041c#diff-455f8729f27403b483bf84b6e8bdcc5afe0ea696ba56209ffcbc7141507cebb3R470-R489) function. I'll switch to using that in a follow up.

> I'm curious what would happen if I sent two messages at the same time with the same conversationID e.g. does one fail or does one hang or is it undefined?

No blocking or failing 😄 
When a user sends a message, we read the conversation history from the ConversationMessage DDB table. If an assistant response hasn't yet been written to that table for a separate in-flight user message, that other in-flight user message isn't included in history.

Reply:
Thanks for clarification on `contentBlockIndex` and double messages for me.